### PR TITLE
TF distributed with NCCL

### PR DESCRIPTION
**Pull request by remicres**
Use NCCL to speed up coms

### CODE REVIEW

**Comment by remicres** [comment]
Maybe we have to update the slurm script to load some additional modules (cuda with nccl?)

File: docs/examples/tf/tf_distributed/mnist_example.py [hunk 1]
```diff
@@ -17,7 +17,9 @@ def train_dense_model(batch_size):
     tf.keras.backend.clear_session()  # For easy reset of notebook state.
 
     slurm_resolver = tf.distribute.cluster_resolver.SlurmClusterResolver(port_base=15000)
-    mirrored_strategy = tf.distribute.MultiWorkerMirroredStrategy(cluster_resolver=slurm_resolver)
+    communication = tf.distribute.experimental.CommunicationImplementation.NCCL
+    mirrored_strategy = tf.distribute.MultiWorkerMirroredStrategy(cluster_resolver=slurm_resolver, 
+                                                                  communication=communication)
```
Comment (zaccharieramzi) [inline docs/examples/tf/tf_distributed/mnist_example.py:22]: I think the kwarg name is `communication_options` for tf 2.4.0.
```suggestion
                                                                  communication_options=communication)
```

**Review by zaccharieramzi — CHANGES_REQUESTED** [review]
Regarding loading nccl, I think it's already loaded with tf 2.4 on JZ

**Comment by RemiLacroix-IDRIS** [comment]
> Maybe we have to update the slurm script to load some additional modules (cuda with nccl?)

All required modules (CUDA, NCCL, etc) should be loaded automatically by our TensorFlow and Pytorch modules.

File: docs/examples/tf/tf_distributed/mnist_example.py (hunk 1 above)
Comment (remicres) [reply docs/examples/tf/tf_distributed/mnist_example.py:22]: Hi @zaccharieramzi , looking at the [source code](https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/python/distribute/collective_all_reduce_strategy.py), it seems you are right. I was fooled by the [the TF 2.4.1 doc](https://www.tensorflow.org/api_docs/python/tf/distribute/experimental/MultiWorkerMirroredStrategy), that mentions `communication`...

Comment (zaccharieramzi) [reply docs/examples/tf/tf_distributed/mnist_example.py]: Yes it fooled me too before, so that's why I was wary of this

**Review by remicres — COMMENTED** [review]
ok

**Review by zaccharieramzi — APPROVED** [review]
LGTM

**Comment by lesteve** [comment]
Thanks a lot merging!

**Pull request merged**